src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedString = '';
+    let newSeparator = separator;
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1) {
+        newSeparator = '';
+      }
+
+      if (this[i] === undefined || this[i] === null) {
+        joinedString += newSeparator;
+      } else {
+        joinedString += this[i] + String(newSeparator);
+      }
```

Comment:
Could be simplified using [Nullish coalescing operator (??)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing).